src/java.base/share/classes/javax/crypto/CipherOutputStream.java

```diff
@@ -91,11 +91,17 @@ public class CipherOutputStream extends FilterOutputStream {
      * Ensure obuffer is big enough for the next update or doFinal
      * operation, given the input length {@code inLen} (in bytes)
      *
+     * If obuffer is null/zero-sized, do not allocate a new buffer.
+     * This reduces allocation for AEAD ciphers that never return data from update
```

Comment:
AEAD ciphers do return data for update() calls for encryption. Perhaps we should add "when used for decryption" or some other similar wordings to the above sentence? Same goes for the comment in CipherInputStream class.

Reply:
Perhaps it's the way you read the sentence, when I read it bit before your comment, I interpreted the change as an open-ended comment where AEAD may or may not return without being specific.  I'm neutral to changing it to specify encryption vs decryption.

Reply:
That's a good suggestion; I changed the wording now. What do you think?

Reply:
Look good, thanks!